src/bloqade/pyqrack/qasm2/core.py

```diff
@@ -77,3 +80,18 @@ def creg_eq(
             return (False,)
 
         return (all(left is right for left, right in zip(lhs, rhs)),)
+
+    @interp.impl(glob.UGate)
+    def ugate(self, interp: PyQrackInterpreter, frame: interp.Frame, stmt: glob.UGate):
```

Comment:
This method needs to be implemented in a method table registered by `glob.dialect`. The convention is to put a method table inside a module with the same name as the dialect.